model.py

```diff
@@ -254,6 +256,30 @@ def compute_mutation_counts(ts):
     return MutationCounts(num_parents, num_inheritors, num_descendants)
 
 
+# Initialize disk cache
+cache = Cache("cache")
+
+
+# The decorator function
+def disk_cache(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        uuid = self.file_uuid
+        if uuid is None:
+            logger.info(f"No uuid, not caching {func.__name__}")
+            return func(self, *args, **kwargs)
+        key = f"{self.file_uuid}-{func.__name__}"
```

Comment:
How will we manage invalidating the cache when the format of the dataframes change?

We could do something like
```
 @disk_cache("v1")
 def summary_df(self):
     ...
```
and put the version string into the key? It takes a bit of management, but seems like a reasonable compromise?

Reply:
We could put the hash of the `tsqc` source in the key? Would save having to remember to change the version number?

Reply:
Yes, could do that. I guess there's pros and cons - if you hash the entire module, then the cache is invalidated for every update, whereas if you just hash the decorated function's code then subtle things like functions being called from that function being changed could lead to the cache being invalid.

Something to follow up on I guess, don't need to work out fully here.